### ISSUE TITLE
about nyu dataset result

### ISSUE THREAD

**Issue by UEA-GCG** [zhwzhong/DCNAS#2]
Hello, Thank you for your work!

After downloading the DCNAS RESULT.zip file provided in the repository, I independently evaluated the NYUv2 ×4 results. However, the RMSE I obtained differs from the value of 1.21 reported in the paper, and I have so far been unable to identify the reason.

My evaluation results are:
===== Without Border Crop =====
RMSE: 1.317335

===== with Border Crop =====
Border: 6 pixels
Cropped shape: torch.Size([449, 1, 468, 628])
RMSE: 1.346227

For the RMSE calculation, I simplified the torch_rmse function provided in the source code:
def torch_rmse(pred, gt, mask, mul_ratio=100):
    valid = mask == 1.
    diff = gt.float()[valid] - pred.float()[valid]
    rmse = torch.sqrt(torch.mean(diff ** 2)).item() * mul_ratio
    return {'RMSE': rmse}

I also checked the original NYUv2 data loader and noticed that the dataset is loaded from:  self.file = h5py.File( '{}/{}.h5'.format(root_path, args.dataset.lower()) )[attr]
Since the provided results contain 449 predictions, I suspect that there may be some differences in the NYUv2 ground-truth preprocessing, data split, or evaluation protocol.

Could you please provide the nyu.h5 file used to generate the reported results, or clarify how the NYUv2 validation/test ground truth was prepared and split?

This would help me reproduce the reported RMSE and better understand the evaluation protocol.

Thank you very much for your help!

**Comment by zhwzhong** [comment]
- Results and GT: https://pan.baidu.com/s/1uF195GgDFDXsoQQOG0rqnQ?pwd=xgx6 提取码: xgx6 
- RMSE calculation code: https://github.com/cvlab-yonsei/dkn

**Comment by UEA-GCG** [comment]
Thank you for your prompt response！
I find that the previously implemented RMSE function has some mistakes: it should calculate the RMSE for each image first and then averaged across all images.
Interestingly, I obtained RMSE values ​​slightly lower than those reported in the paper:
x4: 1.186, 
x8: 2.52, 
x16: 4.83
I have attached my code below in case you have some time to take a look at it. I would be very grateful.
Thank you again!
[evaluate_nyu_files.py](https://github.com/user-attachments/files/32416341/evaluate_nyu_files.py)

**Comment by zhwzhong** [comment]
Thank you for your interest. The provided results were obtained from a retrained model and therefore differ slightly from those reported in the paper. Due to randomness in architecture search and model training, the results may vary across runs. You may use either set of results for comparison.
